Src/main.c

```diff
@@ -108,8 +123,23 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    // Will range from 0 - 4095
+    adcValue = HAL_ADC_GetValue(&hadc);
+
+    adcPercentage = (float)adcValue/MAXADC;
+
+    //Needs to be a value between 1 and 2ms so the it will be the percentage + 1 (50% is really 1.5ms)
+    //That value needs to then be scalled by the timer period and desired period = 60000/20 = 3000
+    pwmValue = (adcPercentage*(HIGH - LOW) + LOW) * SCALE + ROUNDVAL;
```

Comment:
I am curious where the ROUNDVAL is coming from? Do you mind explaining your rationale for adding it?

Reply:
Ya for sure. So currently pwmValue is an integer which means when I solve for it, it will truncate the value and round down. So if the float value for ((adcPercentage*(HIGH - LOW) + LOW) * SCALE) is 3000.7 converting that value to an int (pwmValue) will round down to 3000 even though the value should be 3001. So by adding ROUNDVAL (0.5), it takes this truncation error into account. since if the decimal value of the float is 0.5 or greater adding another 0.5 then rounding that value down will be the equivalent of rounding the value normally. The same is true if the decimal for the value is less than 0.5. Since adding another 0.5 will not change the truncated value. Let me know if you need any more explanation, thanks.